src/views.js

```diff
@@ -204,6 +204,61 @@ const ChoiceChatboxMessageView = new Lang.Class({
     }
 });
 
+
+const PlaceholderOverlay = new Lang.Class({
+    Name: 'PlaceholderOverlay',
+    Extends: Gtk.Overlay,
+    Properties: {
+        placeholder: GObject.ParamSpec.string('placeholder',
+                                              '',
+                                              '',
+                                              GObject.ParamFlags.READWRITE |
+                                              GObject.ParamFlags.CONSTRUCT_ONLY,
+                                              ''),
+        text_view: GObject.ParamSpec.object('text-view',
+                                            '',
+                                            '',
+                                            GObject.ParamFlags.READWRITE |
+                                            GObject.ParamFlags.CONSTRUCT_ONLY,
+                                            Gtk.TextView)
+    },
+
+    _updatePlaceholderState: function() {
+        if (this.text_view.has_focus &&
+            this.text_view.buffer.text) {
```

Comment:
Use `gtk_text_buffer_get_char_count()`, which does not require to allocate a new string